orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java

```diff
@@ -107,4 +109,37 @@ public LLMChoice getChoice() {
         .getChoices()
         .get(0);
   }
+
+  /**
+   * Transform a JSON response into an entity of the given type.
+   *
+   * <p>This is possible on a request with a {@link OrchestrationTemplate#withJsonSchemaResponse}
+   * configured into {@link OrchestrationModuleConfig#withTemplateConfig}.
+   *
+   * @param type the class type to deserialize the JSON content into.
+   * @return the deserialized entity of type T.
+   * @param <T> the type of the entity to deserialize to.
+   */
+  @Nonnull
+  public <T> T entity(@Nonnull final Class<T> type) {
+    String refusal =
+        ((LLMModuleResultSynchronous) getOriginalResponse().getOrchestrationResult())
+            .getChoices()
+            .get(0)
+            .getMessage()
+            .getRefusal();
+    if (refusal != null) {
+      // https://platform.openai.com/docs/guides/structured-outputs?api-mode=responses#refusals
+      throw new OrchestrationClientException(
+          "The model refused to answer the question: " + refusal);
+    }
+    try {
+      return new ObjectMapper().readValue(getContent(), type);
+    } catch (JsonProcessingException e) {
+      throw new OrchestrationClientException(
```

Comment:
(Major)

According to BLI, before throwing the exception we want to log the payload for which the deserialization failed.

Reply:
I added it to the exception